CMakeLists.txt

```diff
@@ -40,17 +59,20 @@ file(GLOB_RECURSE sources CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/source/
 # ---- Create library ----
 
 # Note: for header-only libraries change all PUBLIC flags to INTERFACE and create an interface
-# target: add_library(Greeter INTERFACE) set_target_properties(Greeter PROPERTIES
-# INTERFACE_COMPILE_FEATURES cxx_std_17)
+# target: add_library(Greeter INTERFACE)
+add_library(Greeter)
+target_compile_features(Greeter PUBLIC cxx_std_17)
 
-add_library(Greeter ${headers} ${sources})
-
-set_target_properties(Greeter PROPERTIES CXX_STANDARD 17)
+target_sources(Greeter PRIVATE ${headers} ${sources})
 
 # being a cross-platform target, we enforce standards conformance on MSVC
-target_compile_options(Greeter PUBLIC "$<$<BOOL:${MSVC}>:/permissive->")
+if(MSVC)
+  target_compile_options(Greeter PUBLIC /permissive)
+endif()
 
-# Link dependencies (if required) target_link_libraries(Greeter PUBLIC cxxopts)
+# Link dependencies (if required)
+target_link_libraries(Greeter PRIVATE $<BUILD_INTERFACE:fmt::fmt-header-only>)
+# XXX target_link_libraries(Greeter PUBLIC fmt::fmt)
```

Comment:
I'd prefer to simply link against the `fmt` library here as it avoids the problems described above.

Reply:
on the other hand it shows how to work with an header only library only needed at build time.

Reply:
Yes there are a lot of special cases that would be beneficial for some users. Imo it would be better to document those in the [wiki](https://github.com/TheLartians/ModernCppStarter/wiki), then to add extra complexity to the template. The template should show only the general case and be easy to adapt for other libraries.

Reply:
OK